src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs

```diff
@@ -410,7 +411,7 @@ public IActionResult GetHistory([FromQuery] WalletHistoryRequest request)
 
             try
             {
-                WalletHistoryModel model = WalletModelBuilder.GetHistory(this.walletManager, this.network, request);
+                WalletHistoryModel model = WalletModelBuilder.GetHistorySlim(this.walletManager, this.network, request);
```

Comment:
Wait was that not already using slim? or is it a method the blazor ui is not using?

Reply:
It was not using slim .. with the slim the calculations are now okay

Reply:
This must be tested and verified properly, we've had issues with problematic / different listings of wallet history with this change earlier, or is the problem only now? Either way, will verify.

Reply:
The problem with the right amounts have been there for a while (at least that's what I recall) .. Do you recall what was the problem with the slim? In our case it has been working

Reply:
The issue with slim was that the json structure was different, so it was a breaking change afaik. The current Blockcore Hub (unreleased) appears to function fine with the API now.